btsieve/src/ethereum/web3_http_blocksource.rs

```diff
@@ -61,7 +71,70 @@ impl BlockSource for Web3HttpBlockSource {
                             _ => Err(error)
                         }
                     })
-                    .map_err(Error::Source)
+                    .map_err(Error::Web3)
+            })
+            .filter_map(|maybe_block| maybe_block)
+            .inspect(|block| {
+                if let Block { hash: Some(hash), number: Some(number), .. } = block {
+                    log::trace!(target: "ethereum::blocksource", "latest block is {:?} at height {}", hash, number);
+                }
+            });
+
+        let stream = stream
+            .map(move |block| {
+                block
+                    .transactions
+                    .into_iter()
+                    .filter(|tx| query.matches(&tx))
+                    .collect::<Vec<Self::Transaction>>()
+            })
+            .map(futures::stream::iter_ok)
+            .flatten();
+
+        Box::new(stream)
+    }
+}
+
+impl MatchingTransactions<EventQuery> for Web3HttpBlockSource {
+    type Error = Error;
+    type Transaction = ethereum_support::Transaction;
+
+    fn matching_transactions(
+        &self,
+        query: EventQuery,
+    ) -> Box<dyn Stream<Item = Self::Transaction, Error = Self::Error> + Send + 'static> {
+        let poll_interval = match self.network {
```

Comment:
More relevant to the bitcoin one (can't find it) but I think it would be more readable to warp the match inside the `Duration::from_secs` so that it's clear these figures are seconds. (again, it's more for the bitcoin one which is every few hundred thousand seconds)